t/lib/OpenQA/SeleniumTest.pm

```diff
@@ -337,7 +338,7 @@ sub wait_until {
             return 0;
         }
         $timeout -= $check_interval;
-        wait_for_ajax(msg => $check_description) or sleep $check_interval;
```

Comment:
the idea here was to also shortcut the sleep in case we already waited for "ajax" so there is no need to waste additional sleep time.

Reply:
that unfortunately does not cut it because in `first_iteration` we would still call sleep. How about the following:

```perl
my $do_sleep = 1;
while (1) {
    …
    $timeout -= $check_interval;
    $do_sleep && wait_for_ajax(msg => $check_description) && $do_sleep = 0;
    $do_sleep && sleep $check_interval;
    $do_sleep = 1;
```

Reply:
This reads to me like this:
- Wait for jQuery
  - Unset sleep flag if we have jQuery
- Sleep if we set the flag before
- Reset the flat

The point was to wait for jQuery once. This seems to result in the same behavior.

Reply:
yeah, also wrong. I don't know, a mob programming session today to solve this? ;)